src/components/DiffingResult/index.jsx

```diff
@@ -60,39 +66,141 @@ function DiffingResult() {
 
+  const handleFrameClick = ev => {
+    ev.preventDefault();
+
+    setFrameId(ev.target.getAttribute("data-id"));
+    setFrameName(ev.target.getAttribute("data-name"));
+  };
+
   useEffect(() => {
-    if (diffingResult) {
-      if (diffingResult.result === "error") {
-        setToast({ status: true, message: diffingResult.message });
+    const initCanvas = () => {
+      const canvasInit = new fabric.Canvas("canvas", {
+        width: window.innerWidth,
+        height: window.innerHeight,
+        backgroundColor: "#CED4DA",
+        setZoom: 0.3,
+        selection: false,
+      });
+
+      return canvasInit;
+    };
+
+    const newCanvas = initCanvas();
+    newCanvas.on("mouse:wheel", ev => {
+      const delta = ev.e.deltaY;
+      let Zoom = canvasRef.current.getZoom();
+
+      Zoom *= 0.999 ** delta;
+      Zoom = Math.max(0.01, Math.min(20, Zoom));
+
+      canvasRef.current.zoomToPoint({ x: ev.e.offsetX, y: ev.e.offsetY }, Zoom);
+
+      ev.e.preventDefault();
+      ev.e.stopPropagation();
+    });
 
-        return;
+    canvasRef.current = newCanvas;
+
+    const resizeCanvas = () => {
+      if (canvasRef.current) {
+        canvasRef.current.setHeight(window.innerHeight - 90);
+        canvasRef.current.setWidth(window.innerWidth - 290);
+
+        canvasRef.current.calcOffset();
+        canvasRef.current.renderAll();
       }
+    };
+
+    window.addEventListener("resize", resizeCanvas);
+
+    const fetchFigmaImage = async () => {
+      const urlObject = await fetchImageUrl(projectKey, setToast);
 
-      const frames = [];
+      setImageUrl(urlObject);
+    };
 
-      for (const frameId in diffingResult.content.frames) {
-        const frame = diffingResult.content.frames[frameId];
+    fetchFigmaImage();
 
-        frames.push({ name: frame.name, id: frameId });
+    return () => {
+      window.removeEventListener("resize", resizeCanvas);
+
+      if (canvasRef.current) {
+        canvasRef.current.dispose();
       }
+    };
+  }, []);
+
+  useEffect(() => {
+    if (frameList.length > 0) {
+      const firstFrame = frameList[0];
 
-      setFrameList(frames);
+      setFrameId(firstFrame.id);
+      setFrameName(firstFrame.name);
     }
-  }, [diffingResult]);
+  }, [frameList]);
 
   useEffect(() => {
-    const newCanvas = new fabric.Canvas("canvas", {
-      width: 650,
-      height: 650,
-      backgroundColor: "#CED4DA",
-    });
+    if (!diffingResult) {
+      return;
+    }
 
-    setCanvas(newCanvas);
-  }, []);
+    if (diffingResult.result === "error") {
+      setToast({ status: true, message: diffingResult.message });
 
-  const handleFrameSelect = (frameId, frameName) => {
-    setSelectedFrame({ id: frameId, name: frameName });
-  };
+      return;
+    }
+
+    const frames = [];
+
+    for (const frameId in diffingResult.content.frames) {
+      const frame = diffingResult.content.frames[frameId];
+
+      frames.push({ name: frame.name, id: frameId });
+    }
```

Comment:
- `P2`
`hasOwnPropery`를 사용하면 예상하지 못한 사이드 이펙트를 방지 할 수 있습니다!

```suggestion
    for (const frameId in diffingResult.content.frames) {
      if (diffingResult.content.frames.hasOwnProperty(frameId)) {
        const frame = diffingResult.content.frames[frameId];

        frames.push({ name: frame.name, id: frameId });
      }
    }
```